library/alloc/src/sync.rs

```diff
@@ -237,6 +237,7 @@ macro_rules! acquire {
 /// [rc_examples]: crate::rc#examples
 #[cfg_attr(not(test), rustc_diagnostic_item = "Arc")]
 #[stable(feature = "rust1", since = "1.0.0")]
+#[rustc_insignificant_dtor]
 pub struct Arc<
```

Comment:
It seems that this should also apply to `Weak`? After all, the only thing that dropping a `Weak` might do is freeing the `Arc` and dropping the contained value.

Reply:
Yeah. This PR is not meant to be complete; I just marked types that I saw were missing from the fallout of the crater runs I linked here. I'd rather merge this in this state than modify it further.